### PR TITLE
fix/add tests for validation and ISO 8601

### PR DESCRIPTION
**Pull request by seth-shaw-asu**
**GitHub Issue**: #103 

# What does this Pull Request do?

Addresses issue with date intervals throwing a warning when passed to `EDTF:Utils::iso8601Value`. Also updates and adds more tests.

* Does this change require documentation to be updated? No.
* Does this change add any new dependencies? No.
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? No.
* Could this change impact execution of existing code? No.

# How should this be tested?

Short version: tests pass. :+1: 

Longer version:

* Spin up a site with controlled_access_terms.
* using the command-line: `drush ev 'print(\Drupal\controlled_access_terms\EDTFUtils::iso8601Value("2023-07/2023-08")."\n");'`
* Observe the warning.
* Apply PR
* Run the command again and no warning appears.

# Additional Notes
An early version of this was accidentally pushed directly to the repo which I then reverted. This PR is the real version.

# Interested parties
@Islandora/committers

### CODE REVIEW

**Review by alxp — APPROVED** [review]

**Comment by alxp** [comment]
Tested the drush command and phpunit on Drupal 10, works as advertised.
